### ISSUE TITLE
Speed up rebuilds with `docker build --squash`

### ISSUE THREAD

**Issue by BlaineEXE** [ceph/ceph-container#913]
TL;DR: I propose that once `--squash` is no longer experimental we modify our build process to use it so we can take advantage of more cached layers with only a single resultant layer.

At time of writing, `docker build --squash` is an experimental option. The squash option will take all of the layers added atop the base and squash them so that they appear to be a single layer. So for an image that normally looks like
```
FROM BASE
ADD files
RUN package-manager install ...
ADD more files
```
The squashed image will appear as
```
FROM BASE
(some stuff that's all squashed)
```

This should be able to speed up rebuilds of containers (like for CI), because we can then start to separate out the package manager install from cleanup, etc. We can use as many layers as we like for the images since they will end up squished to as small as possible by the end. By using more layers, we can take advantage of cached layers from previous builds. For example, if we craft our Dockerfile as below, most of the builds can take advantage of the two run operations which are cached, and all the build needs to do is add the modified ceph-container files. And because of the squash option, there won't be the more-layers-adds-more-overhead-to-image-sizes issue since there will appear to only be one resultant layer.
```
FROM base
RUN package-manager update, install, clean
RUN clean operations
ADD ceph-container files
```

I believe there will be a need to force a clean rebuild periodically so that the cached layers have the most updated packages. This can probably be done on weekly-ish basis for CI. Once squash is no longer experimental, we should really consider using it.
